src/frontend/src/flows/authorize/fetchDelegation.ts

```diff
@@ -21,11 +23,18 @@ export const fetchDelegation = async (
   const sessionKey = Array.from(authContext.authRequest.sessionPublicKey);
 
   // at this point, derivationOrigin is either validated or undefined
-  const derivationOrigin =
+  let derivationOrigin =
     authContext.authRequest.derivationOrigin !== undefined
       ? authContext.authRequest.derivationOrigin
       : authContext.requestOrigin;
 
+  // In order to give dapps a stable principal regardless whether they use the legacy (ic0.app) or the new domain (icp0.io)
+  // we map back the derivation origin to the ic0.app domain.
+  const matches = ORIGIN_MAPPING_REGEX.exec(derivationOrigin);
+  if (matches !== null) {
+    derivationOrigin = `https://${matches[1]}.ic0.app`;
```

Comment:
how about using a named group, or at the very least move the `matches[1]` out of the string interpolation (just to clarify any potential errors)